### PR TITLE
Get rid of OPL emulations other than NukedOPL

### PR DESCRIPTION
**Pull request by johnnovak**
Implements https://github.com/dosbox-staging/dosbox-staging/issues/1723

Just removing stuff and some by-the-numbers refactoring & cleanup.
Also renaming `adlib.cpp|h` to `opl.cpp|h`.

### CODE REVIEW

File: src/misc/setup.cpp [hunk 1]
```diff
@@ -703,8 +703,8 @@ bool Section_prop::HandleInputline(string const& gegevens){
 			continue;
 
 		if (p->IsDeprecated()) {
-			LOG_MSG("CONFIG: Deprecated option '%s'", name.c_str());
-			LOG_MSG("CONFIG: %s", p->GetHelp());
+			LOG_WARNING("CONFIG: Deprecated option '%s'", name.c_str());
+			LOG_WARNING("CONFIG: %s", p->GetHelp());
```
Comment (kcgen) [inline src/misc/setup.cpp:707]: very helpful!

File: src/hardware/mixer.cpp [hunk 2]
```diff
@@ -1760,7 +1760,7 @@ void init_mixer_dosbox_settings(Section_prop &sec_prop)
 	auto int_prop = sec_prop.Add_int("rate", only_at_start, default_rate);
 	assert(int_prop);
 	const char *rates[] = {
-	        "8000", "11025", "16000", "22050", "32000", "44100", "48000", "49716", 0};
+	        "8000", "11025", "16000", "22050", "32000", "44100", "48000", 0};
```
Comment (kcgen) [inline src/hardware/mixer.cpp:1770]: :+1:

File: src/hardware/opl.cpp [hunk 3]
```diff
@@ -31,254 +31,294 @@
...
+	skipWrite:
+		// Not yet capturing to a file here
+		// Check for commands that would start capturing, if it's not
+		// one of them return
+		if (!(
+		            // note on in any channel
+		            (regMask >= 0xb0 && regMask <= 0xb8 && (val & 0x020)) ||
+		            // Percussion mode enabled and a note on in any
+		            // percussion instrument
+		            (regMask == 0xbd && ((val & 0x3f) > 0x20)))) {
```
Comment (kcgen) [inline src/hardware/opl.cpp]: `clang-format` struggled with this one!
maybe temps would help.. 
``` c++
const auto is_note_on_any_ch = regMask >= 0xb0 && regMask <= 0xb8 && (val & 0x020);
const auto is_perc_on_with_note = regMask == 0xbd && ((val & 0x3f) > 0x20));
if (!is_note_on_any_ch && !is_perc_on_with_note)
	return;
```
(or something a bit more digestible)

File: src/hardware/opl.h [hunk 4]
```diff
@@ -59,87 +58,86 @@ class Timer {
...
 			// How far into the next cycle
 			const double deltaTime = time - trigger;
 			// Sync start to last cycle
 			const auto counterMod = fmod(deltaTime, counterInterval);
-			start = time - counterMod;
-			trigger = start + counterInterval;
-			//Only set the overflow flag when not masked
+			start                 = time - counterMod;
+			trigger               = start + counterInterval;
+			// Only set the overflow flag when not masked
```
Comment (kcgen) [inline src/hardware/opl.h]: Maybe an empty line separating the two constants versus the two members (logically separate their different roles, and the right-alignment would tighten up for `start` and `trigger`).

File: src/hardware/opl.h [hunk 5]
```diff
@@ -152,47 +150,52 @@ class OPL: public Module_base {
...
 
 	RegisterCache cache;
-	Capture* capture;
-	Chip	chip[2];
+	Capture *capture;
+	Chip chip[2];
 
 	opl3_chip oplchip = {};
-	uint8_t newm = 0;
+	uint8_t newm      = 0;
```
Comment (kcgen) [inline src/hardware/opl.h:113]: Might as well toss in defaults for all the members.
Reviewing commit-by-commit (please ignore if a later commit takes care of this :slightly_smiling_face: )

File: src/hardware/opl.cpp (hunk 3 above)
Comment (johnnovak) [reply src/hardware/opl.cpp]: 😅 I agree

**Comment by johnnovak** [comment]
> Regression tested this branch w/ Dune + AdlibGold (using our prior conf), but it doesn't progress past the Cryo logo screen and I'm not seeing the usual debug messages.
> 
> ```ini
> [sblaster]
> sbtype = none
> sbmixer = false
> oplmode = opl3gold
> opl_filter = auto
> ```

Same thing here; I'll look into it.

**Comment by kcgen** [comment]
@johnnovak ,

In the filter-disabled logs, would it be OK to exclude the `(0 dB/oct at 0 Hz)` metrics? 

``` text
2022-06-20 19:30:23.941 | MIXER: FM channel operating at 48000 Hz without resampling
2022-06-20 19:30:23.942 | MIXER: FM channel lowpass filter disabled (0 dB/oct at 0 Hz)
2022-06-20 19:30:23.942 | SBPRO2: OPL filter disabled
2022-06-20 19:30:23.943 | MIXER: SB channel operating at 22050 Hz and upsampling to the output rate
2022-06-20 19:30:23.943 | MIXER: SB channel lowpass filter disabled (0 dB/oct at 0 Hz)
2022-06-20 19:30:26.072 | SBPRO2: DSP was reset
2022-06-20 19:30:26.083 | SBPRO2: Speaker-output has been toggled on
2022-06-20 19:30:26.242 | MIXER: SB channel lowpass filter enabled (0 dB/oct at 0 Hz)
```

**Comment by johnnovak** [comment]
> @johnnovak ,
> 
> In the filter-disabled logs, would it be OK to exclude the `(0 dB/oct at 0 Hz)` metrics?
> 
> ```
> 2022-06-20 19:30:23.941 | MIXER: FM channel operating at 48000 Hz without resampling
> 2022-06-20 19:30:23.942 | MIXER: FM channel lowpass filter disabled (0 dB/oct at 0 Hz)
> 2022-06-20 19:30:23.942 | SBPRO2: OPL filter disabled
> 2022-06-20 19:30:23.943 | MIXER: SB channel operating at 22050 Hz and upsampling to the output rate
> 2022-06-20 19:30:23.943 | MIXER: SB channel lowpass filter disabled (0 dB/oct at 0 Hz)
> 2022-06-20 19:30:26.072 | SBPRO2: DSP was reset
> 2022-06-20 19:30:26.083 | SBPRO2: Speaker-output has been toggled on
> 2022-06-20 19:30:26.242 | MIXER: SB channel lowpass filter enabled (0 dB/oct at 0 Hz)
> ```

Doesn't this commit do the trick?
https://github.com/dosbox-staging/dosbox-staging/pull/1738/commits/1c6eb2029b620d30b508bbf967a873b24a3e4cdd

**Comment by kcgen** [comment]
> Doesn't this commit do the trick? https://github.com/dosbox-staging/dosbox-staging/commit/1c6eb2029b620d30b508bbf967a873b24a3e4cdd

Perfectly! Nice and tidy :-)

**Comment by johnnovak** [comment]
Okay, I'm done with this for now, it's in a somewhat better shape I think.

I've fixed the regression, a single line got deleted somehow 10 commits ago, that was all.

**Comment by FeralChild64** [comment]
One thing - shouldn't `opl3gold` be compatible with `opl3`? With SB16 configured and `oplmode` set to `opl3gold` software in general is unable to play OPL3 music any more. This includes Dune configured to use SoundBlaster.

**Comment by johnnovak** [comment]
> @johnnovak One thing - shouldn't `opl3gold` be compatible with `opl3`? With SB16 configured and `oplmode` set to `opl3gold` software in general is unable to play OPL3 music any more. This includes Dune configured to use SoundBlaster.

Good catch, thanks for reporting it! The `Reset()` on init got lost somehow during the final rebase. Fixed in https://github.com/dosbox-staging/dosbox-staging/pull/1742

**Pull request merged**